pkg/objects/swift.go

```diff
@@ -104,40 +134,69 @@ func (s SwiftLocation) Validate(name string) []error {
 
 var accountCache = map[string]*schwift.Account{}
 
+type logger struct {
+	Prefix string
+}
+
+func (l logger) Printf(format string, args ...interface{}) {
+	for _, v := range strings.Split(fmt.Sprintf(format, args...), "\n") {
+		logg.Debug("[%s] %s", l.Prefix, v)
+	}
+}
+
 //Connect implements the Source interface. It establishes the connection to Swift.
-func (s *SwiftLocation) Connect() error {
+func (s *SwiftLocation) Connect(name string) error {
 	if s.Account != nil {
 		return nil
 	}
 
 	//connect to Swift account (but re-use connection if cached)
-	key := s.cacheKey()
+	key := s.cacheKey(name)
 	s.Account = accountCache[key]
 	if s.Account == nil {
-		authOptions := gophercloud.AuthOptions{
-			IdentityEndpoint: s.AuthURL,
-			Username:         s.UserName,
-			DomainName:       s.UserDomainName,
-			Password:         string(s.Password),
-			Scope: &gophercloud.AuthScope{
-				ProjectName: s.ProjectName,
-				DomainName:  s.ProjectDomainName,
-			},
-			AllowReauth: true,
+		authInfo := &clientconfig.AuthInfo{
```

Comment:
Why this change from `gophercloud.AuthOptions` to `clientconfig.AuthInfo`? AuthOptions already supports application credentials.

Reply:
application credentials don't require scope, I don't want to introduce yet another `if then else` logic, it's better to reuse the existing package.

Reply:
I think it's better to use a small if-then-else than to pull in another entire module.

Reply:
this module already handles all the possible auth conditions and well tested. if new auth case is added, it is easier just to bump the dependency or set an extra parameter.

Reply:
All that clientconfig does is produce an AuthOptions struct anyway. As a big proponent of the YAGNI principle (You Ain't Gonna Need It), I reverted this change before merging to keep the implementation simple.